### PR TITLE
feat: created floating button ui component for add and mic

### PR DESCRIPTION
**Pull request by PreyanshK**
This is the Floating Action button that will be re-used for multiple screens to add a new list item or open the Mic.

Add New Item:
image

Mic:
image

Screen Position:
image

### CODE REVIEW

File: app/src/main/java/com/example/farmeraid/uicomponents/models/UiComponentModel.kt [hunk 1]
```diff
@@ -21,4 +21,17 @@ class UiComponentModel {
...
+    // Card List Models
+    data class CardUiState(
+        val text : String = ""
+    )
+
+    // Floating Action Button Models
+    data class FabUiState(
+        val icon: String,
+    )
+    data class FabUiEvent(
```
Comment (adshayanB) [inline app/src/main/java/com/example/farmeraid/uicomponents/models/UiComponentModel.kt:34]: Nice!

**Review by adshayanB — APPROVED** [review]
Don't forget to add the time log :)

**Review by RishanR — CHANGES_REQUESTED** [review]
The overall structure is looking good. Honestly, I think instead of passing in a string, we could just pass in an Icons object. That way this view can be used for any icon. So we could just pass in something like Icons.Rounded.Add into the FabUiState. You could still force the tint to white in the view, and add an optional contentDescription parameter in the FabUiState.

**Comment by adshayanB** [comment]
> The overall structure is looking good. Honestly, I think instead of passing in a string, we could just pass in an Icons object. That way this view can be used for any icon. So we could just pass in something like Icons.Rounded.Add into the FabUiState. You could still force the tint to white in the view, and add an optional contentDescription parameter in the FabUiState.

This is smart, it allows for scalability to support other icons if we need.

**Comment by PreyanshK** [comment]
> The overall structure is looking good. Honestly, I think instead of passing in a string, we could just pass in an Icons object. That way this view can be used for any icon. So we could just pass in something like Icons.Rounded.Add into the FabUiState. You could still force the tint to white in the view, and add an optional contentDescription parameter in the FabUiState.

Yeah, that's a cleaner implementation. I'll fix it.

File: app/src/main/java/com/example/farmeraid/uicomponents/FloatingActionButtonView.kt [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import com.example.farmeraid.ui.theme.WhiteContentColour
+import com.example.farmeraid.uicomponents.models.UiComponentModel
+
+@Composable
+fun FloatingActionButtonView(
+    fabUiState: UiComponentModel.FabUiState,
+    fabUiEvent: UiComponentModel.FabUiEvent,
+    modifier: Modifier = Modifier,
+) {
+    Box(modifier = Modifier.fillMaxSize()) {
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/uicomponents/FloatingActionButtonView.kt:30]: Also another quick question. Is there a reason why we need to wrap this around a box and add the `fillMaxSize` modifier?

File: app/src/main/java/com/example/farmeraid/uicomponents/models/UiComponentModel.kt [hunk 3]
```diff
@@ -21,4 +23,19 @@ class UiComponentModel {
...
+
+    // Card List Models
+    data class CardUiState(
+        val text : String = ""
+    )
+
+    // Floating Action Button Models
+    data class FabUiState(
+        val icon: ImageVector,
+        val contentDescription: String = "",
```
Comment (RishanR) [inline app/src/main/java/com/example/farmeraid/uicomponents/models/UiComponentModel.kt:35]:
```suggestion
        val contentDescription: String? = null,
```

Comment (RishanR) [reply app/src/main/java/com/example/farmeraid/uicomponents/models/UiComponentModel.kt:35]: Just a minor change nit pick

**Review by RishanR — APPROVED** [review]
Everythings looking good! Just a minor nit pick that I commented on. Otherwise, you can just rebase main and resolve the TimeLog conflict and merge!

Great work!

**Pull request merged**